pkg/controllers/vscr/scrutinizepod_reconciler.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright [2021-2023] Open Text.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vscr
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	v1 "github.com/vertica/vertica-kubernetes/api/v1"
+	"github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/builder"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	verrors "github.com/vertica/vertica-kubernetes/pkg/errors"
+	"github.com/vertica/vertica-kubernetes/pkg/vscrstatus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+// ScrutinizePodReconciler will reconcile for the scrutinize
+// pod creation
+type ScrutinizePodReconciler struct {
+	VRec *VerticaScrutinizeReconciler
+	Vscr *v1beta1.VerticaScrutinize
+	Log  logr.Logger
+}
+
+// MakeScrutinizePodReconciler will build a ScrutinizePodReconciler object
+func MakeScrutinizePodReconciler(r *VerticaScrutinizeReconciler, vscr *v1beta1.VerticaScrutinize,
+	log logr.Logger) controllers.ReconcileActor {
+	return &ScrutinizePodReconciler{
+		VRec: r,
+		Vscr: vscr,
+		Log:  log,
+	}
+}
+
+func (s *ScrutinizePodReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctrl.Result, error) {
+	isSet := s.Vscr.IsStatusConditionTrue(v1beta1.ScrutinizePodCreated)
+	if isSet {
+		return ctrl.Result{}, nil
+	}
+
+	// collect information from a VerticaDB.
+	if res, err := s.collectInfoFromVdb(ctx); verrors.IsReconcileAborted(res, err) {
+		return res, err
+	}
+
+	pod := builder.BuildScrutinizePod(s.Vscr)
+	s.Log.Info("Creating scrutinize pod", "Name", s.Vscr.ExtractNamespacedName())
+	err := ctrl.SetControllerReference(s.Vscr, pod, s.VRec.Scheme)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	err = s.VRec.Client.Create(ctx, pod)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	s.Log.Info("Scrutinize pod created successfully")
+	err = vscrstatus.UpdateCondition(ctx, s.VRec.Client, s.Vscr,
+		v1.MakeCondition(v1beta1.ScrutinizePodCreated, metav1.ConditionTrue, "PodCreated"))
```

Comment:
Were we going to update the pod name and pod uid in the status field too?

Reply:
On this PR I was going to update just the condition, but yes I think it makes sense to update those fields here too